web-js/src/components/WelcomePage.js

```diff
@@ -0,0 +1,146 @@
+import React from "react";
+import * as Realm from "realm-web";
+import {
+  Container,
+  TextField,
+  Button,
+  IconButton,
+  Card,
+  Typography,
+  InputAdornment,
+} from "@material-ui/core";
+import { useRealmApp } from "./RealmApp";
+import { MoreInfoTemplateAndDocs } from './MoreInfo'
+import Visibility from "@material-ui/icons/Visibility";
+import VisibilityOff from "@material-ui/icons/VisibilityOff";
+import { toggleBoolean } from "../utils";
+
+
+
+export function WelcomePage() {
+  const realmApp = useRealmApp();
+  // Track whether the user is logging in or signing up for a new account
+  const [isSignup, setIsSignup] = React.useState(false);
+  const toggleIsSignup = () => {
+    clearErrors();
+    setIsSignup(toggleBoolean);
+  };
+  // Authentication errors
+  const noErrors = {
+    email: null,
+    password: null,
+  };
+  const [error, setError] = React.useState(noErrors);
+  const clearErrors = () => setError(noErrors);
+  // Manage password visibility
+  const [showPassword, setShowPassword] = React.useState(false);
+  const toggleShowPassword = () => setShowPassword(toggleBoolean);
+
+  const onFormSubmit = async ({ email, password }) => {
+    clearErrors();
+    try {
+      if (isSignup) {
+        await realmApp.emailPasswordAuth.registerUser(email, password);
+      }
+      await realmApp.logIn(Realm.Credentials.emailPassword(email, password));
+    } catch (err) {
+      handleAuthenticationError(err, setError);
+    }
+  };
+
+  return (
+    <Container maxWidth="sm" className="main-container">
+      <Card className="auth-card" variant="outlined">
+        <form
+          className="auth-form"
+          onSubmit={(e) => {
+            e.preventDefault();
+            function getFormValues(htmlFormElement) {
+              const formData = new FormData(htmlFormElement);
+              return Object.fromEntries(formData.entries());
+            }
+            const { email, password } = getFormValues(e.target);
```

Comment:
Seems a little excessive to create an inline function for this.
```suggestion
            const formData = new FormData(e.target);
            const { email, password } = Object.fromEntries(formData.entries());
```